cmd/collect/main.go

```diff
@@ -170,30 +171,32 @@ func runCollector(cCtx *cli.Context) error {
 	}
 
 	// Start service components
-	opts := collector.CollectorOpts{
-		Log:                     log,
-		UID:                     uid,
-		Location:                location,
-		OutDir:                  outDir,
-		CheckNodeURI:            checkNodeURI,
-		ClickhouseDSN:           clickhouseDSN,
-		Nodes:                   nodeURIs,
-		BloxrouteAuth:           blxAuth,
-		EdenAuth:                edenAuth,
-		ChainboundAuth:          chainboundAuth,
-		Receivers:               receivers,
-		ReceiversAllowedSources: receiversAllowedSources,
-		APIListenAddr:           apiListenAddr,
-		MetricsListenAddr:       metricsListenAddr,
-		EnablePprof:             enablePprof,
-	}
-
-	collector.Start(&opts)
+	collector := collector.New(collector.CollectorOpts{
+		Log:                      log,
+		UID:                      uid,
+		Location:                 location,
+		OutDir:                   outDir,
+		CheckNodeURI:             checkNodeURI,
+		ClickhouseDSN:            clickhouseDSN,
+		Nodes:                    nodeURIs,
+		BloxrouteAuth:            blxAuth,
+		EdenAuth:                 edenAuth,
+		ChainboundAuth:           chainboundAuth,
+		Receivers:                receivers,
+		ReceiversAllowedSources:  receiversAllowedSources,
+		ReceiversAllowAllSources: len(receiversAllowedSources) == 1 && receiversAllowedSources[0] == "all",
```

Comment:
Should we move this logic to `NewTxProcessor` and not propagate new field everywhere in opts?

Reply:
done